ofctrl/fgraphFlow.go

```diff
@@ -843,12 +843,13 @@ func (self *Flow) installFlowActions(flowMod *openflow13.FlowMod,
 
 			log.Debugf("flow install. Added setTunnelId Action: %+v", setTunnelAction)
 
-		case "setMetadata":
-			// Set Metadata instruction
+		case ActTypeSetMetadata:
```

Comment:
Actually, I am a bit confused. I know you want to add support for applyAction set metadata, so you add a new OFAction with it, but here you keep the original write_metadata instruction. 

With your current code, if using `SetMetadataAction.GetActionMessage`, an ApplyAction message is generated, and can be add into ApplyActionInstruction, but if call `Flow.installFlowActions` it uses write_metadata instruction.

My point is, please keep the behavior compatible, maybe you should rename the original action type as write_meteadata, and add a new type set_metadata, and add call in both `SetMetadataAction.GetActionMessage`, and `Flow.installFlowActions`.